admiral/envs/components/movement.py

```diff
@@ -11,53 +11,29 @@ class GridMovementAgent(Agent):
     move (int):
         The maximum number of cells away that the agent can move.
     """
-    def __init__(self, move=None, **kwargs):
+    def __init__(self, move_range=None, **kwargs):
         super().__init__(**kwargs)
-        assert move is not None, "move must be an integer"
-        self.move = move
+        assert move_range is not None, "move_range must be an integer"
+        self.move_range = move_range
     
     @property
     def configured(self):
         """
         Agents are configured if the move parameter is set.
         """
-        return super().configured and self.move is not None
+        return super().configured and self.move_range is not None
 
-class GridMovementComponent:
-    """
-    Agents can move around in the grid. This component processes the move action.
-    If the new position is out of bounds, then the position is not updated.
-
-    The agents' action space is appended with Box(-agent.move, agent.move, (2,), np.int),
-    indicating that the agent can move in 2 dimensions up to its maximum distance.
-
-    region (int):
-        The size of the region. This is needed to determine if the agent has attempted
-        to move out of bounds or not.
-
-    agents (dict):
-        The dictionary of agents. Because the movement is based on the agent's
-        posiiton, the agent must be GridPositionAgents.
-    """
-    def __init__(self, region=None, agents=None, **kwargs):
-        assert type(region) is int, "Region must be an integer"
-        self.region = region
-        assert type(agents) is dict, "agents must be a dict"
-        for agent in agents.values():
-            assert isinstance(agent, GridPositionAgent)
+class GridMovementActor:
+    def __init__(self, position=None, agents=None, **kwargs):
+        self.position = position
         self.agents = agents
 
         from gym.spaces import Box
         for agent in self.agents.values():
             if isinstance(agent, GridMovementAgent):
-                agent.action_space['move'] = Box(-agent.move, agent.move, (2,), np.int)
+                agent.action_space['move'] = Box(-agent.move_range, agent.move_range, (2,), np.int)
```

Comment:
If the agent is a move agent, then it must be a position agent. Assert position agent in this loop